applications/luci-app-xinetd/htdocs/luci-static/resources/view/xinetd/xinetd.js

```diff
@@ -0,0 +1,211 @@
+		o.modalonly = false;
+
+		o = s.option(form.DummyValue, 'protocol', _('Protocol'));
+		o.modalonly = false;
+
+		o = s.option(form.DummyValue, 'server', _('Server'));
+		o.modalonly = false;
+
+		o = s.option(form.DummyValue, 'disable', _('Enabled'));
+		o.cfgvalue = function(section) {
+			return (uci.get('xinetd', section, 'disable') == "no") ? _("yes") : _("no");
+		};
+		o.modalonly = false;
+
+		s.tab('basic', _('Basic Settings'));
+		s.tab('advanced', _('Advanced Settings'));
+
+		// Now here follow the "real" values to be set in the modal (with the hint texts)
+
+		// Basic settings
+		o = s.taboption('basic', form.Flag, 'disable', _('Enabled'), _('Enable or Disable this service'));
+		o.enabled  = 'no';
+		o.disabled = 'yes';
+		o.default  = o.enabled;
+		o.rmempty = false;
+		o.modalonly = true;
+
+		o = s.taboption('basic', form.ListValue, 'type', _('Type'), _('Type of service'));
+		o.default = 'UNLISTED';
+		// FIXME for now we will only support unlisted services, maybe later we could use the (very long) list from /etc/services if needed
+		// o.value('INTERNAL', _('INTERNAL'));
+		o.value('UNLISTED', _('UNLISTED'));
+		o.rmempty = false;
+		o.modalonly = true;
+
+		o = s.taboption('basic', form.Value, 'port', _('Port'), _('The port used for this service, valid range: 0 - 65535'));
+		o.datatype = 'port';
+		o.depends('type', 'UNLISTED');
+		o.rmempty = false;
+		o.modalonly = true;
+		o.validate = function(section_id, value) {
+			var sections = uci.sections('xinetd', 'service');
+
+			for (var i = 0; i < sections.length; i++) {
+				if (uci.get('xinetd', sections[i]['.name'], 'port') == value && section_id != sections[i]['.name'])
+					return _('Port already in use by service "%s"'.format(sections[i]['.name']));
+			}
+
+		return true;
+		};
+
+		o = s.taboption('basic', form.ListValue, 'wait', _('Threading behaviour'), _('Selection of the threading for this service'));
+		o.default = 'no';
+		o.value('yes', _('Single-Threaded Service'));
+		o.value('no', _('Multi-Threaded Service'));
+		o.rmempty = false;
+		o.modalonly = true;
+
+		o = s.taboption('basic', form.ListValue, 'socket_type', _('Socket type'), _('The type of the socket used for this service'));
+		o.default = 'stream';
+		o.value('stream', _('stream-based service'));
+		o.value('dgram', _('datagram-based service'));
+		o.value('raw', _('direct access to IP service'));
+		o.value('seqpacket', _('sequential datagram transmission service'));
+		o.rmempty = false;
+		o.modalonly = true;
+
+		o = s.taboption('basic', form.ListValue, 'protocol', _('Protocol'), _('The protocol to be used for this service'));
+		o.default = 'tcp';
+		o.value('tcp', _('TCP'));
+		o.value('udp', _('UDP'));
+		o.rmempty = false;
+		o.modalonly = true;
+
+		o = s.taboption('basic', form.ListValue, 'user', _('User (UID)'), _('User ID for the server process for this service'));
+		o.rmempty = false;
+		o.modalonly = true;
+		o.load = function(section_id) {
```

Comment:
I suggest to use `widgets.UserSelect` here. This way you can get rid of the custom `load()` and `getUsers()` logic.

Reply:
Thank you for your hint. I will modify it.
But, do you have a tipp how to clear the (existing) keylist if not clearing it in a custom load()?
The list get's longer every time as the user list is appended to it.
Thanks in advance.

Reply:
That is probably a bug we should fix separately in `widgets.js`

Reply:
OK.
What do you mean, should I wait for the fix or maybe doing a workaround wit a load() function which resets the list by doing a `this.keylist = 0` only (as I already did)